### PR TITLE
Refactor test data for `cellranger` modules

### LINKED ISSUE #867: refactor test data for `cellranger` modules
in writing a `cellranger multi` module (https://github.com/nf-core/modules/pull/3229), it became evident that the corresponding test data PR (https://github.com/nf-core/test-datasets/pull/848) doesn't fully test the module

there are two options:
* simply adding more data (easiest)
* refactoring tests for the extant `cellranger` modules (`count`, `mkgtf`, `mkref`, `vdj`, `mkvdjref`, `mkfastq`, and `multi`) to rely on a single data store

the problem with simply adding data is that it bloats the repo

10X furnishes a few multiomic datasets that we could incorporate:
* [Integrated GEX, TotalSeq™-C, and TCR Analysis of Chromium Connect Generated Library from 5k CMV+ T cells](https://www.10xgenomics.com/resources/datasets/integrated-gex-totalseqc-and-tcr-analysis-of-connect-generated-library-from-5k-cmv-t-cells-2-standard)
* [Human PBMC from a Healthy Donor, 10k cells - multi (v2)](https://www.10xgenomics.com/resources/datasets/human-pbmc-from-a-healthy-donor-10-k-cells-multi-v-2-2-standard-5-0-0)
* [30k Mouse E18 Combined Cortex, Hippocampus and Subventricular Zone Cells Multiplexed, 12 CMOs](https://www.10xgenomics.com/resources/datasets/30-k-mouse-e-18-combined-cortex-hippocampus-and-subventricular-zone-cells-multiplexed-12-cm-os-3-1-standard-6-0-0)

in a refactor, we could pick and choose which data to use for each module, e.g. `cellranger count` would rely on one of the GEX datasets, `cellranger vdj` on one of the immune profiling datasets, etc.

 `cellranger multi` would test all datasets

the original datasets include full FASTQs, but previous `cellranger` tests have successfully downsampled them for nf-core module testing

### PR DESCRIPTION
**Pull request by klkeys**
closes #867

major refactor of 10X Genomics Cellranger test data to accommodate adequate testing for a new `cellranger multi` module (https://github.com/nf-core/modules/pull/3229)

test data now all now sit within a `cellranger/` folder under `data/genomics/homo_sapiens/illumina/10xgenomics/`

test data for other *ranger software (`cellranger-arc`, `cellranger-atac`, `spaceranger`) could be easily as sister folders

### CODE REVIEW

File: data/genomics/homo_sapiens/illumina/10xgenomics/cellranger/10k_pbmc_cmo/10k_pbmc_cmo_config.csv [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[gene-expression]
+reference,/references/deleteme
```
Comment (sateeshperi) [inline data/genomics/homo_sapiens/illumina/10xgenomics/cellranger/10k_pbmc_cmo/10k_pbmc_cmo_config.csv:2]: is this path with deleteme intentional?

**Comment by sateeshperi** [comment]
m guessing this will also need an update to `modules/tests/config/test_data.config` once this is approved

**Comment by jfy133** [comment]
I sort of feel this shouldn't be embedded in `illumina/` as it's even if it uses Illumina sequencing, it is a separate platform I guess, but rather alongside. Secondly it's not derived from the main chr22 (or chr21? don't remember now) dataset like the vast majority of the other data in that folder, so by pushing it to a sister folder that will be slightly clearer. But that's my personal opinion.

Otherwise the data is small, and well documented with their own personal READMEs (which isn't the standard, but I think it's helpful here when we are dealing with prop. software).

Only thing missing is the short descriptions on the main repo README (on the modules branch).

File: data/genomics/homo_sapiens/illumina/10xgenomics/cellranger/10k_pbmc_cmo/10k_pbmc_cmo_config.csv (hunk 1 above)
Comment (klkeys) [reply data/genomics/homo_sapiens/illumina/10xgenomics/cellranger/10k_pbmc_cmo/10k_pbmc_cmo_config.csv:2]: oops, that's an artifact from when I tested the subsampled data before creating the PR

`deleteme` is simply the name of the reference, I'll change it to something more realistic

Comment (sateeshperi) [reply data/genomics/homo_sapiens/illumina/10xgenomics/cellranger/10k_pbmc_cmo/10k_pbmc_cmo_config.csv:2]: i might have seen it in some other file as well

**Comment by klkeys** [comment]
>m guessing this will also need an update to modules/tests/config/test_data.config once this is approved
@sateeshperi I intend to handle it in https://github.com/nf-core/modules/pull/3229, since I also need to point all `cellranger` module tests to the new or adjusted data paths anyways

it will be a pretty big fix 😵 but eminently doable

> by pushing it to a sister folder that will be slightly clearer. But that's my personal opinion.
@jfy133 I have no strong feelings either way

I think that all these 10X test data were sequenced on Illumina platforms, but they definitely aren't derived from chr21/chr22

I'll move these data to `data/genomics/homo_sapiens/10xgenomics` (sister to `illumina`)

File: data/genomics/homo_sapiens/illumina/10xgenomics/cellranger/10k_pbmc_cmo/10k_pbmc_cmo_config.csv (hunk 1 above)
Comment (klkeys) [reply data/genomics/homo_sapiens/illumina/10xgenomics/cellranger/10k_pbmc_cmo/10k_pbmc_cmo_config.csv]: you definitely did! it was in two of the `cellranger multi` config files

very good eye 👌🏼

Comment (klkeys) [reply data/genomics/homo_sapiens/illumina/10xgenomics/cellranger/10k_pbmc_cmo/10k_pbmc_cmo_config.csv]: `deleteme` is now deleted 👍🏼

**Review by sateeshperi — APPROVED** [review]
Thanks for the detailed README's

**Comment by jfy133** [comment]
Thanks for moving @klkeys ! You're missing still the update to the main data descritopr section of the README:

https://github.com/nf-core/test-datasets/tree/modules#data-description

CAn ocme in a follow up PR :)

**Comment by klkeys** [comment]
🤦🏼🤦🏼🤦🏼  thank you for keeping me accountable @jfy133 

I'll open that PR shortly 👍🏼

**Comment by klkeys** [comment]
see https://github.com/nf-core/test-datasets/pull/884

**Comment by klkeys** [comment]
thanks for this heads-up @adamrtalbot 

the original 10X GEX FASTQs got replaced, but there are new FASTQs with the same 10x single cell 3' v3 chemistry

did a quick `grep -R "test_10x_1_fastq_gz" *` in this repo, and looks like I also broke kallistobustools/count 😣 

since I broke things, I'll compile fixes and post a PR soon 👍🏼

**Comment by klkeys** [comment]
see here: https://github.com/nf-core/modules/pull/3486

**Pull request merged**